Source/WebCore/page/scrolling/ScrollingTreeNode.h

```diff
@@ -107,7 +107,7 @@ class ScrollingTreeNode : public ThreadSafeRefCounted<ScrollingTreeNode> {
     const ScrollingNodeType m_nodeType;
     const ScrollingNodeID m_nodeID;
 
-    ScrollingTreeNode* m_parent { nullptr };
```

Comment:
Why do you use `ThreadSafeWeakPtr` rather than `CheckedPtr` for `m_parent`?
Weak pointers allocates another memory block for `ThreadSafeWeakPtrControlBlock`.
The original code was using a raw pointer for `m_parent` because it doesn't need a weak pointer.

Reply:
Because CheckedPtr is not thread safe, and there isn't really a way to avoid weak-ness when thread safety is required.

Reply:
We have `CanMakeThreadSafeCheckedPtr`.
If a `CanMakeThreadSafeCheckedPtr` object is destructed in another thread, it fails the release assertion that checks the atomic counter is zero.
thread-safety doesn't bother in this case.

Reply:
Oh right, I totally forgot that I added that. I have reasons to believe that weak semantics is needed in this instance, unfortunately.